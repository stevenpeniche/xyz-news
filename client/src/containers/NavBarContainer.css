.nav-bar {
  display: flex;
  position: fixed;
  align-items: center;
  height: 45px;
  width: 100%;
  background-color: #000000;
  box-shadow: 0 1px 2px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  z-index: 1;
}

#logo {
  order: 1;
  margin-left: 20px;
  margin-right: 20px;
  font-family: 'Audiowide', sans-serif;
  font-size: 35px;
  letter-spacing: -2px;
  color: #ffffff;
  text-decoration: none;
}

.nav-bar-menu {
  order: 2;
}

.nav-bar-menu-icon {
  margin-top: 5px;
  font-size: 26px;
  color: #ffffff;
  text-decoration: none;
  animation-delay: .4s;
}

.nav-bar-menu-icon:hover {
  cursor: pointer;
  color: #ffffffb7;
}

.nav-bar-menu-list-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  position: fixed;
  top: 22px;
  left: 250px;
  z-index: 2;
  min-width: 120px;
  background-color: #ffffff;
  -moz-border-radius: 8px;
	-webkit-border-radius: 8px;
	border-radius: 8px;
  box-shadow: 0 3px 6px rgba(0,0,0,0.16), 0 3px 6px rgba(0,0,0,0.23);
  overflow: hidden;
  animation-duration: .4s;
}

.nav-bar-menu-list {
  margin-bottom: 10px;
}

.nav-bar-menu-list-item {
  text-align: center;
  text-decoration: none;
  color: #000000;
}

.menu-item-source-icon-container {
  margin-top: 10px;
}

.scroll-to-top-section {
  order: 3;
  flex-grow: 1;
  height: 100%;
  margin-left: 20px;
}

.scroll-to-top-section:hover {
  cursor: pointer;
}

/* Mobile Devices */
@media only screen and (max-width: 414px) {
  #logo {
    font-size: 30px;
    margin-left: 14px;
    margin-right: 0;
  }

  .nav-bar-menu {
    order: 4;
    margin-right: 14px;
  }

  .nav-bar-menu-list-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: fixed;
    top: 30px;
    left: auto;
    right: 45px;
    max-height: 320px;
    overflow-y: auto;
  }
}

/* Landscape Mobile Devices */
@media only screen and (max-width: 812px) and (orientation: landscape){
  #logo {
    font-size: 30px;
    margin-left: 14px;
    margin-right: 0;
  }

  .nav-bar-menu {
    order: 4;
    margin-right: 14px;
  }

  .nav-bar-menu-list-container {
    display: flex;
    flex-direction: column;
    align-items: center;
    position: fixed;
    top: 30px;
    left: auto;
    right: 45px;
    max-height: 320px;
    overflow-y: auto;
  }
}