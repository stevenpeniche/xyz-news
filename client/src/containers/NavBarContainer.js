import React, {Component} from 'react';
import { Link } from 'react-router-dom';
import { animateScroll as scroll } from 'react-scroll';
import shortid from 'shortid';
import './NavBarContainer.css';

class NavBar extends Component {
  state = {
    menuActive: false
  }

  scrollToTop() {
    scroll.scrollToTop();
  }

  setMenuStatus = (target) => {
    this.setState((prevState) => {
      const nextState = target === 'logo' ? false : !prevState.menuActive
      return {menuActive: nextState}
    });
  }

  render() {
    const visibility = this.state.menuActive ? "visible" : "hidden";
    const menuAnimationClass = this.state.menuActive ? "animated fadeIn" : "";
    return (
      <nav className="nav-bar">
        <Link to="/"
          id="logo"
          onClick={(e) => this.setMenuStatus(e.target.id)}
        >
          XYZ news
        </Link>
        {this.props.currentSourceID ?
          <div className="nav-bar-menu">
            <div className="nav-bar-menu-icon animated fadeInLeft" onClick={this.setMenuStatus}>
              <i className="menu-icon fas fa-bars"></i>
            </div>
            <div className={`nav-bar-menu-list-container ${menuAnimationClass}`} style={{"visibility": visibility}}>
              <div className="nav-bar-menu-list">
                {this.props.sources.map((source) => {
                  return(
                    <Link to={`/${source.id}`} key={shortid.generate()}
                      className="nav-bar-menu-list-item"
                      onClick={this.setMenuStatus}>
                      <div className="menu-item-source-icon-container">
                        <img
                          className="source-icon levitate"
                          src={source.faviconURL}
                          alt={source.title}
                          style={{backgroundColor: source.primaryColor}}
                        />
                      </div>
                      <div className="source-title">
                        {source.title}
                      </div>
                    </Link>
                  )
                })}
              </div>
            </div>
          </div>
          :
          <div></div>
        }
        <div className="scroll-to-top-section"  onClick={this.scrollToTop}>
        ...
        </div>
      </nav>
    );
  }
}

export default NavBar;