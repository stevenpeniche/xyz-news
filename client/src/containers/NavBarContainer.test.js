import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import NavBarContainer from './NavBarContainer';

let currentSourceID, sources, setMenuStatus, wrapper;

beforeEach(() => {
  currentSourceID = null;

  sources = [
    {
      id: 'bleacher-report',
      title: 'Bleacher Report',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#000000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'MLB'
        },
        {
          name: 'NFL'
        },
        {
          name: 'NBA'
        },
        {
          name: 'UFC'
        },
        {
          name: 'Boxing'
        },
        {
          name: 'WWE'
        }
      ]
    },
    {
      id: 'cnn',
      title: 'CNN',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#cc0000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'World'
        },
        {
          name: 'Politics'
        },
        {
          name: 'Money'
        },
        {
          name: 'Health'
        },
        {
          name: 'Travel'
        },
        {
          name: 'Tech'
        }
      ]
    },
    {
      id: 'vice-news',
      title: 'Vice News',
      faviconURL: 'https://besticon-demo.herokuapp.com/icon?url=https://bleacherreport.com/&size=70..120..200',
      primaryColor: '#000000',
      topics: [
        {
          name: 'Top'
        },
        {
          name: 'Drugs'
        },
        {
          name: 'Tech'
        },
        {
          name: 'Money'
        },
        {
          name: 'Immigration'
        },
        {
          name: 'Politics'
        },
        {
          name: 'Terrorism'
        }
      ]
    }
  ];

  wrapper =
    shallow(
      <NavBarContainer
        currentSourceID={currentSourceID}
        sources={sources}
      />
    );
});

describe('NavBarContainer rendering', () => {
  it('renders without crashing', () => {
    wrapper
  });

  it('renders correctly', () => {
    const tree =
      renderer.create(
        <MemoryRouter>
          <NavBarContainer
            currentSourceID={currentSourceID}
            sources={sources}
          />
        </MemoryRouter>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('initial state', () => {
  it('menuActive', () => {
    expect(wrapper.state().menuActive).toBe(false);
  });
});

describe('#setMenuStatus', () => {
  it('sets activeMenu state to it\'s inverse', () => {
    expect(wrapper.state().menuActive).toBe(false);
    wrapper.instance().setMenuStatus()
    expect(wrapper.state().menuActive).toBe(true);
  });

  it('doesn\'t set activeMenu state if target is logo', () => {
    expect(wrapper.state().menuActive).toBe(false);
    wrapper.instance().setMenuStatus('logo')
    expect(wrapper.state().menuActive).toBe(false);
  });
});