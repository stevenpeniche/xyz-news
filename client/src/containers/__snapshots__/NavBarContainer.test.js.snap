// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBarContainer rendering renders correctly 1`] = `
<nav
  className="nav-bar"
>
  <a
    href="/"
    id="logo"
    onClick={[Function]}
  >
    XYZ news
  </a>
  <div />
  <div
    className="scroll-to-top-section"
    onClick={[Function]}
  >
    ...
  </div>
</nav>
`;
